pkg/operations/start.go

```diff
@@ -56,15 +57,33 @@ func (start StartOpsHandler) ActionStartedCondition(reqCtx intctrlutil.RequestCt
 func (start StartOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
 	var (
 		cluster   = opsRes.Cluster
-		startComp = func(compSpec *appsv1.ClusterComponentSpec) {
-			compSpec.Stop = nil
-		}
+		startList = opsRes.OpsRequest.Spec.StartList
 	)
-	for i := range cluster.Spec.ComponentSpecs {
-		startComp(&cluster.Spec.ComponentSpecs[i])
+	compOpsHelper := newComponentOpsHelper(startList)
+	// abort earlier running opsRequests.
+	if err := abortEarlierOpsRequestWithSameKind(reqCtx, cli, opsRes, []opsv1alpha1.OpsType{opsv1alpha1.StopType},
+		func(earlierOps *opsv1alpha1.OpsRequest) (bool, error) {
+			if len(startList) == 0 {
+				// start all components
+				return true, nil
+			}
+			return len(earlierOps.Spec.StopList) == 0 || hasIntersectionCompOpsList(compOpsHelper.componentOpsSet, earlierOps.Spec.StopList), nil
```

Comment:
What if the previous ops stops A and B, but the current ops starts only B, should the A be stopped?

Reply:
yes, A will be stopped by cluster controller.  and only the stop ops will be aborted.